src/sm/pmix_mmap.c

```diff
@@ -58,12 +60,21 @@ int _mmap_segment_create(pmix_sm_seg_t *sm_seg, const char *file_name, size_t si
         goto out;
     }
     /* size backing file - note the use of real_size here */
+#ifdef HAVE_POSIX_FALLOCATE
+    if (0 != posix_fallocate(sm_seg->seg_id, 0, size)) {
+        pmix_output_verbose(2, pmix_globals.debug_output,
+                "sys call posix_fallocate(2) fail\n");
+        rc = PMIX_ERROR;
```

Comment:
I suggest to check the `errno` value for clarity.
Calling `posix_fallocate` returns `ENOSPC` in  the case where the storage space ended. So can will check it and return the PMIx like error code: `PMIX_ERR_OUT_OF_RESOURCE`
This allows to see the root cause of failure.

It can look like:
```C
switch( errno ) {
    case ENOSPC:
        rc = PMIX_ERR_OUT_OF_RESOURCE;
        break;
    default:
        rc = PMIX_ERROR;
        break;
}
```